File: Leetcode Daily Challenge/013-July-2024/001-three-consecutive-odds.java
// https://leetcode.com/problems/three-consecutive-odds/



class Solution {
    // TC : O(N)
    // SC : O(1)
    public boolean threeConsecutiveOdds(int[] arr) {
        for (int i = 0; i + 2 < arr.length; i ++) {
            if ((arr[i]     & 1) == 1 && 
                (arr[i + 1] & 1) == 1 && 
                (arr[i + 2] & 1) == 1) {
                return true;
            }
        }
        return false;
    }
}

File: Leetcode Daily Challenge/013-July-2024/001-three-consecutive-odds.py
# https://leetcode.com/problems/three-consecutive-odds/



class Solution(object):
    # TC : O(N)
    # SC : O(1)
    def threeConsecutiveOdds(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        for i in range(len(arr) - 2):
            if arr[i] & 1 and arr[i + 1] & 1 and arr[i + 2] & 1:
                return True
        return False